### PR TITLE
remove unhelpful quotes to fix bug with reading key from a path with …

### LINKED ISSUE #8599: Vagrant Builder does not find SSH private key in vagrant.d directory if there is a space in the path
#### Overview of the Issue

I'm not sure if this is related to issue #8413, but a colleague and I have observed that if there is a space in the path to the Vagrant home directory, then Packer seems to get an SSH error when it tries to look for the `insecure_private_key` file. 

#### Reproduction Steps

This fails, post-provisioning: `PACKER_LOG=1 VAGRANT_HOME="/c/directory with space" packer build -force centos.json`

This does not: `PACKER_LOG=1 VAGRANT_HOME="/c/nospacedir" packer build -force centos.json`

where `directory with space` and `nospacedir` are new, empty directories. 

centos.json:
https://gist.github.com/bensonbenson/72a951ff883da26289b5342397a4f8e2

vagrantfile.template:
https://gist.github.com/bensonbenson/943d3b1e546628a3175459d2732789ff

### Packer version

Packer v1.5.1

### Operating system and Environment details

This is on Windows 10 within the Git Bash shell (MINGW64). A colleague has noticed this is an issue on ArchLinux (not sure the specifics of his setup, but creating a new Vagrant home directory with spaces in its name will cause the same issue.)

### Log Fragments and crash.log files

```
==> virtualbox: Using ssh communicator to connect: 127.0.0.1
==> virtualbox: Waiting for SSH to become available...
2020/01/13 11:41:27 packer.exe plugin: [INFO] Waiting for SSH, up to timeout: 10m0s
2020/01/13 11:41:27 packer.exe plugin: [DEBUG] Error getting SSH config: Error on reading SSH private key: open "C:/space dir/insecure_private_key": The filename, directory name, or volume label syntax is incorrect.
2020/01/13 11:41:32 packer.exe plugin: [DEBUG] Error getting SSH config: Error on reading SSH private key: open "C:/space dir/insecure_private_key": The filename, directory name, or volume label syntax is incorrect.
2020/01/13 11:41:37 packer.exe plugin: [DEBUG] Error getting SSH config: Error on reading SSH private key: open "C:/space dir/insecure_private_key": The filename, directory name, or volume label syntax is incorrect.
```

### PR DESCRIPTION
**Pull request by SwampDragons**
…spaces in it.

Closes #8599

### CODE REVIEW

File: builder/vagrant/driver_2_2.go [hunk 1]
```diff
@@ -199,7 +199,11 @@ func (d *Vagrant_2_2_Driver) vagrantCmd(args ...string) (string, string, error)
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
 
-	cmd.Start()
+	err := cmd.Start()
+	if err != nil {
+		return "", "", fmt.Errorf("Error starting vagrant command with args: %s",
```
Comment (azr) [inline builder/vagrant/driver_2_2.go]: I recently found out about the `%q` option of fmt and it's awesome at quoting strings for us : https://play.golang.org/p/x2R2sNFZyFa

```suggestion
		return "", "", fmt.Errorf("Error starting vagrant command with args: %q",
```

File: builder/vagrant/step_ssh_config.go [hunk 2]
```diff
@@ -53,6 +55,15 @@ func (s *StepSSHConfig) Run(ctx context.Context, state multistep.StateBag) multi
...
 	}
+	log.Printf("identity file is %s", sshConfig.IdentityFile)
+	log.Printf("Removing quotes from identity file")
+	if strings.HasPrefix(sshConfig.IdentityFile, "\"") {
+		sshConfig.IdentityFile = sshConfig.IdentityFile[1:]
+		if strings.HasSuffix(sshConfig.IdentityFile, "\"") {
+			sshConfig.IdentityFile = sshConfig.IdentityFile[:len(sshConfig.IdentityFile)-1]
+		}
+		log.Printf("identity file is now %s", sshConfig.IdentityFile)
+	}
```
Comment (azr) [inline builder/vagrant/step_ssh_config.go]: Here, strconv.Unqote should be our best friend: https://golang.org/pkg/strconv/#Unquote

```go
if s, err := strconv.Unqote(sshConfig.IdentityFile; err == nil {
    sshConfig.IdentityFile = s
}
```

File: builder/vagrant/driver_2_2.go (hunk 1 above)
Comment (azr) [reply builder/vagrant/driver_2_2.go]: Also; nice a new error is handled !!! 💯

**Comment by SwampDragons** [comment]
ah, TIL about Unquote. Definitely the better way to do it.

**Pull request merged**